libraries/chain/sps_objects/sps_processor.cpp

```diff
@@ -0,0 +1,286 @@
+      {
+         auto sum = std::accumulate(   _voter.proxied_vsf_votes.begin(),
+                                       _voter.proxied_vsf_votes.end(),
+                                       _voter.vesting_shares.amount );
+         ret += sum.value;
+      }
+
+      ++found;
+   }
+
+   return ret;
+}
+
+void sps_processor::calculate_votes( const t_proposals& proposals )
+{
+   for( auto& item : proposals )
+   {
+      const proposal_object& _item = item;
+      auto total_votes = calculate_votes( _item.proposal_id );
+
+      db.modify( _item, [&]( auto& proposal )
+                        {
+                           proposal.total_votes = total_votes;
+                        } );
+   }
+}
+
+void sps_processor::sort_by_votes( t_proposals& proposals )
+{ 
+   std::sort( proposals.begin(), proposals.end(), []( const proposal_object& a, const proposal_object& b )
+                                                      {
+                                                         if (a.total_votes == b.total_votes)
+                                                         {
+                                                            return a.proposal_id < b.proposal_id;
+                                                         }
+                                                         return a.total_votes > b.total_votes;
+                                                      } );
+}
+
+asset sps_processor::get_treasury_fund()
+{
+   auto& treasury_account = db.get_account( STEEM_TREASURY_ACCOUNT );
+
+   return treasury_account.sbd_balance;
+}
+
+asset sps_processor::get_daily_inflation()
+{
+   //temporary
+   return asset( 0, SBD_SYMBOL );
+}
+
+asset sps_processor::calculate_maintenance_budget( const time_point_sec& head_time )
+{
+   //Get funds from 'treasury' account ( treasury_fund ) 
+   asset treasury_fund = get_treasury_fund();
+
+   //Get daily proposal inflation ( daily_proposal_inflation )
+   asset daily_inflation = get_daily_inflation();
+
+   FC_ASSERT( treasury_fund.symbol == daily_inflation.symbol, "symbols must be the same" );
+
+   //Calculate budget for given maintenance period
+   auto passed_time_seconds = ( head_time - db.get_dynamic_global_properties().last_budget_time ).to_seconds();
+
+   //Calculate daily_budget_limit
+   auto daily_budget_limit = treasury_fund.amount.value / 100 + daily_inflation.amount.value;
+   daily_budget_limit = daily_budget_limit * ( static_cast< double >( passed_time_seconds ) / daily_seconds );
+
+   //Transfer daily_proposal_inflation to `treasury account`
+   transfer_daily_inflation_to_treasury( daily_inflation );
+
+   return asset( daily_budget_limit, treasury_fund.symbol );
+}
+
+void sps_processor::transfer_daily_inflation_to_treasury( const asset& daily_inflation )
+{
+   if( daily_inflation.amount.value > 0 )
+   {
+      const auto& treasury_account = db.get_account( STEEM_TREASURY_ACCOUNT );
+      db.adjust_balance( treasury_account, daily_inflation );
```

Comment:
Is this printing new SBD?

That's not how we have handled inflation in the past. Either inflation should be paid, per block, in to the treasury account in `database::process_funds` or added to a temp fund in the dgpo that is then transferred in to the treasury account during maintenance.